src/main/java/com/example/demo/assembler/UserAssembler.java

```diff
@@ -0,0 +1,66 @@
+package com.example.demo.assembler;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+import com.example.demo.model.User;
+import com.example.demo.response.dto.UserDto;
+import com.example.demo.response.dto.UserResponseDto;
+
+@Component
+public class UserAssembler {
+	
+	public UserResponseDto toModel(List<User> users, Optional<User> opUser)
+	{
+		UserResponseDto userResponse = new UserResponseDto();
+		UserDto userDto = new UserDto();
+		List<UserDto> userDtolist = new ArrayList<>();
+		
+		if(CollectionUtils.isEmpty(users) && opUser.isPresent()== true)
+		{
+			userDto.setName(opUser.get().getName());
+			userDto.setDesignation(opUser.get().getDesignation());
+			userDto.setDu(opUser.get().getDu());
+			userDto.setExtension(opUser.get().getExtension());
+			userDto.setImg(opUser.get().getImg());
+			userDto.setImgurl(opUser.get().getImgurl());
+			userDto.setMail(opUser.get().getMail());
+			userDto.setOracleid(opUser.get().getOracleid());
+			userDto.setProject(opUser.get().getProject());
+			userDto.setSeat(opUser.get().getSeat());
+			
+			userDtolist.add(userDto);
+			
+		}
+		else if(CollectionUtils.isEmpty(users) ==false && opUser.isPresent() == false)
+		{
+				for(User s: users)
+				{
+					System.out.println(s.getId());
+					
+					userDto.setName(s.getId());
+					userDto.setDesignation(s.getDesignation());
+					userDto.setDu(s.getDu());
+					userDto.setExtension(s.getExtension());
+					userDto.setImg(s.getImg());
+					userDto.setImgurl(s.getImgurl());
+					userDto.setMail(s.getMail());
+					userDto.setOracleid(s.getOracleid());
+					userDto.setProject(s.getProject());
+					userDto.setSeat(s.getSeat());
+					
+					userDtolist.add(userDto);
```

Comment:
Use streams and map function for conversion

Reply:
Used Stream instead. Though has some discrepancy . userDtolist2 is holding the last inserted data only. Incorrect O/P.

`List<UserDto> userDtolist2 = users.parallelStream().map(
				(usr) -> 
				{	
					userDto.setName(usr.getName());
					userDto.setDesignation(usr.getDesignation());
					userDto.setDu(usr.getDu());
					userDto.setExtension(usr.getExtension());
					userDto.setImg(usr.getImg());
					userDto.setImgurl(usr.getImgurl());
					userDto.setMail(usr.getMail());
					userDto.setOracleid(usr.getOracleid());
					userDto.setProject(usr.getProject());
					userDto.setSeat(usr.getSeat());
					return userDto;

				}).collect(Collectors.toList());`